src/ol/style/IconImage.js

```diff
@@ -85,12 +90,15 @@ class IconImage extends EventTarget {
    */
   isTainted_() {
     if (this.tainted_ === undefined && this.imageState_ === ImageState.LOADED) {
```

Comment:
Doesn't this check ensure that there is only one canvas created. Once the `tainted_` is set this will no longer be executed.

Reply:
+1 my last commit can be removed, do you agree @ahocevar?

Reply:
The way you made the last commit does not bring an improvement. I was suggesting to have a static context, not a member context. Just replace `this.taintedTestContext_` with `taintedTestContext`, and initialize that outside the class with
```js
const taintedTestContext = null;
```